pkg/systemd/bootInfo.jsx

```diff
@@ -0,0 +1,138 @@
+import { EmptyState, EmptyStateBody } from "@patternfly/react-core/dist/esm/components/EmptyState";
+
+const _ = cockpit.gettext;
+
+export function BootInfo() {
+    const [svg, setSvg] = useState(undefined);
+    const [summary, setSummary] = useState(null);
+
+    useEffect(() => {
+        let string = "";
+        cockpit.spawn(["systemd-analyze", "plot"])
+                .stream((svg_part) => {
+                    string += svg_part;
+                })
+                .then(() => {
+                    const doc = new DOMParser().parseFromString(string, "text/xml");
+                    doc.querySelector("rect.background")?.remove();
+                    doc.querySelector("text[y='30']")?.remove();
+                    const svgElem = doc.querySelector("svg");
+                    svgElem.style.scale = "0.5";
+                    const summaryElem = doc.querySelector("text[y='50']");
+                    // Grabs summary from SVG and puts it in the card body instead
+                    if (summaryElem !== null) {
+                        setSummary(summaryElem.textContent);
+                        summaryElem.remove();
+                    }
+                    const [plot, legend] = doc.querySelectorAll("g");
+                    legend.remove();
+                    [...plot.querySelectorAll("text.left"), ...plot.querySelectorAll("text.right")].forEach((text) => {
+                        const match = text.innerHTML.match(/^(?<service>.+\.[a-z._-]+)(\s+\((?<time>\d+(\.\d+)?)\w+\))?$/);
+                        if (match !== null) {
+                            const service = match.groups.service;
+                            const time = match.groups.time;
+                            text.setAttribute("data-service", service);
+                            text.setAttribute("data-time", time);
+                            text.classList.add("clickable-service");
+                        }
+                    });
+                    setSvg(doc.documentElement);
+                })
+                .catch((e) => {
+                    setSvg(null);
+                    setSummary(e);
+                });
+    }, []);
+
+    if (svg === undefined) {
+        return (
+            <div className="pf-v5-c-page__main-section">
+                <EmptyState variant={EmptyStateVariant.xs}>
+                    <EmptyStateHeader titleText={_("Loading")} headingLevel="h4" />
+                    <EmptyStateBody>
+                        <Spinner size="xl" />
+                    </EmptyStateBody>
+                </EmptyState>
+            </div>
+        );
+    }
+
+    if (svg === null) {
+        return (
+            <div className="pf-v5-c-page__main-section">
+                <EmptyState variant={EmptyStateVariant.xs}>
+                    <EmptyStateHeader titleText={_("Failure")} headingLevel="h4" />
+                    <EmptyStateBody>
+                        {_("Are you sure systemd-analyze is available?")}
```

Comment:
Let's please drop this sentence. The error message should already point it out, and it's rather hard to *not* have systemd-analyze if you have systemd at all.

(Also, cockpit-components-empyt-state)